Framework/DataHandling/src/LoadILLPolarizedDiffraction.cpp

```diff
@@ -0,0 +1,411 @@
+}
+/**
+ * Runs LoadInstrument as child to link the instrument to workspace
+ */
+void LoadILLPolarizedDiffraction::loadInstrument(
+    API::MatrixWorkspace_sptr &workspace) {
+  IAlgorithm_sptr loadInst = createChildAlgorithm("LoadInstrument");
+  loadInst->setPropertyValue("Filename", getInstrumentFilePath(m_instName));
+  loadInst->setProperty<MatrixWorkspace_sptr>("Workspace", workspace);
+  loadInst->setProperty("RewriteSpectraMap", OptionalBool(true));
+  loadInst->execute();
+}
+
+/**
+ * Loads twotheta for each detector pixel from the file
+ * @param workspace : workspace with loaded instrument
+ * @param entry : entry from which the pixel 2theta positions will be read
+ * @param bankId : bank ID for which 2theta positions will be read
+ * @return : vector of pixel 2theta positions in the chosen bank
+ */
+std::vector<double> LoadILLPolarizedDiffraction::loadTwoThetaDetectors(
+    const API::MatrixWorkspace_sptr &workspace, const NXEntry &entry,
+    int bankId) {
+
+  std::vector<double> twoTheta(static_cast<int>(D7_NUMBER_PIXELS_BANK));
+
+  if (getPropertyValue("PositionCalibration").compare("Nexus") == 0) {
+    NXFloat twoThetaPixels = entry.openNXFloat(
+        "D7/Detector/bank" + std::to_string(bankId) + "_offset");
+    twoThetaPixels.load();
+    for (auto pixelId = 0; pixelId < twoThetaPixels.size(); pixelId++) {
+      twoTheta[pixelId] = twoThetaPixels[pixelId];
+    }
+  } else {
+    IAlgorithm_sptr loadInst = createChildAlgorithm("LoadParameterFile");
+    loadInst->setPropertyValue("Filename", getPropertyValue("YIGFilename"));
+    loadInst->setProperty("Workspace", workspace);
+    loadInst->execute();
+
+    auto instrumentMap = workspace->instrumentParameters();
+    Instrument_const_sptr instrument = workspace->getInstrument();
+
+    for (auto ii = 0; ii < static_cast<int>(D7_NUMBER_PIXELS_BANK); ii++) {
+      auto pixel = instrument->getDetector(ii + 1);
+      std::vector<double> twoThetaRead =
+          (pixel->parameterMap())
+              .getDouble(pixel->getName(),
+                         std::string("twoTheta_bank" + std::to_string(bankId)));
+      twoTheta[ii] = twoThetaRead[0];
+    }
+  }
+  return twoTheta;
+}
+
+/**
+ * Rotates each pixel to its corresponding 2theta read from the file
+ * @param entry : entry from which the 2theta positions will be read
+ * @param workspace : workspace containing the instrument being moved
+ */
+void LoadILLPolarizedDiffraction::moveTwoTheta(
+    const NXEntry &entry, API::MatrixWorkspace_sptr &workspace) {
+
+  Instrument_const_sptr instrument = workspace->getInstrument();
+
+  auto &componentInfo = workspace->mutableComponentInfo();
+  for (auto ii = 0; ii < static_cast<int>(D7_NUMBER_BANKS); ii++) {
+    NXFloat twoThetaBank = entry.openNXFloat(
+        "D7/2theta/actual_bank" +
+        std::to_string(ii + 2)); // detector bank IDs start at 2
+    twoThetaBank.load();
+    if (getPropertyValue("PositionCalibration").compare("None") == 0) {
```

Comment:
I think we understood that the bank positions should be set from nexus in all the cases?

Reply:
And it is implemented as such. In the case where `PositionCalibration` property is set to `None` the whole banks are rotated by the angle read from the NeXuS file, otherwise the code reads the relative pixel position and the bank rotation, and then places each pixel accordingly.